core/src/main/java/org/mule/runtime/core/internal/processor/strategy/BlockingProcessingStrategyFactory.java

```diff
@@ -81,5 +84,26 @@ public ReactiveProcessor onProcessor(ReactiveProcessor processor) {
       }
     }
 
+    /**
+     * This strategy adds a Mono.block call in order to preserve the thread, because it's a precondition to transactions to work.
+     * However, there are some operations that don't need to use a Mono.block, because they have a synchronous execution. It
+     * allows us to make a performance optimization for those operations, and this method is intended to detect which operations
+     * do need a Mono.block and which don't.
+     *
+     * @param processor The processor.
+     * @return true if a Mono.block call is needed to wait for operation completion.
+     */
+    private static boolean needsMonoBlock(ReactiveProcessor processor) {
+      try {
+        // This method is added in the SDK, and it's return value is false for those operations which include a
+        // completion callback. That completion callback is the way to implement a non-blocking operations, so if this
+        // method returns true, we'll need a Mono.block call to wait the operation completion.
+        Method isBlockingMethod = processor.getClass().getDeclaredMethod("isBlocking");
+        isBlockingMethod.setAccessible(true);
+        return !(boolean) isBlockingMethod.invoke(processor);
```

Comment:
This is expensive and being done for every call. Cache this result or precalculate and make it available some other way.

Reply:
Reflection is baaad